lib/dpul_collections/indexing_pipeline/figgy/hydration_cache_entry.ex

```diff
@@ -17,4 +17,61 @@ defmodule DpulCollections.IndexingPipeline.Figgy.HydrationCacheEntry do
     |> cast(attrs, [:data, :cache_version, :record_id, :source_cache_order])
     |> validate_required([:data, :cache_version, :record_id, :source_cache_order])
   end
+
+  @spec to_solr_document(%__MODULE__{}) :: %{}
+  def to_solr_document(hydration_cache_entry) do
+    %{record_id: id} = hydration_cache_entry
+    %{data: %{"metadata" => metadata = %{"title" => title}}} = hydration_cache_entry
+    description = get_in(metadata, ["description"])
+    years = extract_years(metadata)
+    display_date = format_date(metadata)
+
+    %{
+      id: id,
+      title_ss: title,
+      description_txtm: description,
+      years_is: years,
+      display_date_s: display_date
+    }
+  end
+
+  defp extract_years(%{"date_range" => [%{"start" => [start_year], "end" => [end_year]}]}) do
+    Enum.to_list(String.to_integer(start_year)..String.to_integer(end_year))
+  end
+
+  defp extract_years(%{"date_created" => []}) do
+    nil
+  end
+
+  defp extract_years(%{"date_created" => date}) do
+    Enum.map(date, &String.to_integer/1)
```

Comment:
Is this...there are multiple dates sometimes? Should this be `dates`?

Reply:
Yes it should :)

Reply:
actually no I think it shouldn't. It's a single-value field in the figgy form.

Reply:
Okay I expect it will be a single date. But it's stored in an array because figgy does that. And it's going into a multi-valued field. Solr will map it to multi-value even if we give it a single value, but is that easier to understand than just outputting it as an array here to begin with? I'm trying a few things and none of them seem super clear. I may just comment.

Reply:
What about `"date_created" => [date]`?

Reply:
😂 gmta, see if you like it.

Reply:
I do like it! Part of me wonders if `"date_created" => [date | _tail]` or something would more explicitly say "even if there's two dates somehow, that's a mistake and we don't care", but I think this is good.

Reply:
actually that's a good point. I was gonna say we'd get a match error in that case but, but currently if that did happen it would fall through to the next clause and return nil and we'd never know we lost that date.